packages/react-sdk/src/hooks/storage/useCustomUploader.ts

```diff
@@ -1,76 +1,56 @@
 import { useState, useCallback } from 'react';
-import { useArweaveStorage } from './arweave/useArweave';
 import { WalletContextState } from '@solana/wallet-adapter-react';
 import { Connection } from '@solana/web3.js';
 import { SessionWalletInterface } from '../session';
-
-export type UploaderType = 'arweave' | 'genesysgo' | 'custom';
-
-export interface UploaderFunction {
-  (data: string): Promise<string>;
-}
+import { Uploader } from './UploaderInterface';
+import { useArweaveStorage } from './arweave/useArweave';
+import { ArweaveUpload } from './ArweaveUpload';
+import { GenesysGoUpload } from './GenesysGoUpload';
 
 export const useCustomUploader = (
-  uploaderType: UploaderType,
+  uploaderType: 'arweave' | 'genesysgo' | 'custom',
```

Comment:
Why are we letting user specify`'arweave' | 'genesysgo'` in the custom uploader? I guess this is confusing.